docs/userguides/console.md

```diff
@@ -0,0 +1,62 @@
+# Ape Console
+
+Ape provides an [IPython](https://ipython.readthedocs.io/) interactive console with useful pre-defined locals to interact with your project.
+
+```bash
+ape console --network ethereum:mainnet
+
+In [1]: networks.active_provider.get_block("latest")
+Out[1]: Block(gas_data=BlockGasFee(gas_limit=30029267, gas_used=1957793, base_fee=22138196051), consensus_data=BlockConsensus(difficulty=13087655881211628, total_difficulty=43800816864239190379319), hash=HexBytes('0x75af0ab1a0bcd42a6d894574a47d7933dd6d1ad67820245d596bee2464b676cf'), number=14388141, parent_hash=HexBytes('0x75af0ab1a0bcd42a6d894574a47d7933dd6d1ad67820245d596bee2464b676cf'), size=17431, timestamp=1647306023)
+```
+
+## Namespace Extras
+
+You can also create scripts to be included in the console namespace by adding a file (`ape_console_extras.py`) to your root project directory.  All non-internal symbols from this file will be included in the console namespace.  Internal symbols are prefixed by an underscore (`_`).
+
+An example file might look something like this:
+
+```python
+from eth_utils import encode_hex, decode_hex
+
+_networks = None
+
+
+def latest(key="number"):
+    return getattr(_networks.active_provider.get_block("latest"), key)
+
+
+def ape_init_extras(networks):
+    """Setup _networks so util func can access it"""
+    global _networks
```

Comment:
Is using a global something you've been looking to do?

Reply:
It was intentional.  It allows the function to alter `_networks` that gets included in the namespace.  

It could be argued this is questionable style and could be structured differently to return a func in the dict from `ape_init_extras()` but I wanted to include this as an example of altering symbols within the file.

Reply:
I think it might be pretty advanced for an example. These docs are going to be referenced by a lot of newcomers, so I'd like to use simpler things as examples, like maybe importing a popular dependency like matplotlib or Decimal

Reply:
I am not at all familiar with matplotlib.  You do have me worried I'm writing for the wrong audience though.  I can simplify my example, but I'm not sure I can write for other types of users very well.  Simplifying this example might look like:

    from eth_utils import encode_hex, decode_hex


    def ape_init_extras(networks):
        """Setup _networks so util func can access it"""
        return {
            "latest": lambda key: getattr(networks.active_provider.get_block("latest"), key)
        }

I could create another example to parse token transfer logs and drop value of Decimal but that's probably equally not simple (`print('Hello, World')`?).  I'll have to mull it over but open to suggestions.